services/blockchain-connector/tests/unit/fs.test.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * LiskHQ/lisk-service
+ * Copyright © 2023 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+const { resolve, dirname } = require('path');
+
+const {
+	exists,
+	extractTarBall,
+	mkdir,
+	read,
+	rm,
+} = require('../../shared/utils/fs');
+
+const { genesisBlock } = require('../constants/blocks');
+
+describe('Unit tests for filesystem utilities', () => {
+	const dirPath = resolve(`${dirname(__dirname)}/testDir`);
+	const filePath = resolve(`${process.cwd()}/tests/constants/genesis_block.json`);
+	const tarFilePath = resolve(`${process.cwd()}/tests/constants/genesis_block.json.tar.gz`);
+
+	afterAll(async () => {
+		// Remove test directory
+		await rm(dirPath, { recursive: true, force: true });
+	});
+
+	it('should create directory', async () => {
+		expect(exists(dirPath)).resolves.toBe(false);
+		await mkdir(dirPath, { recursive: true });
+		expect(exists(dirPath)).resolves.toBe(true);
+	});
+
+	it('should extract tar file', async () => {
+		const outputPath = `${dirPath}/genesis_block.json`;
+		expect(exists(outputPath)).resolves.toBe(false);
+
+		// Extract tar file
+		await extractTarBall(tarFilePath, dirPath);
+		expect(exists(outputPath)).resolves.toBe(true);
+	});
+
+	it('should read data from file', async () => {
+		expect(exists(filePath)).resolves.toBe(true);
+		// Read data from file
+		const result = await read(filePath);
+		const parsedResult = JSON.parse(result);
+		expect(parsedResult).toMatchObject(genesisBlock);
+	});
+});
```

Comment:
Can we add cases when filepath does not exists / is null or undefined or empty string? 

Also, better to put these tests under functional test as these interacts with system. The reason to do this is - in general, the functional / integration tests are costlier and at a point we will need to convert some of these to unit tests. Having tests segregated will help then.